transform/flatten_mangler.go

```diff
@@ -60,11 +63,11 @@ func (f *FlattenMangler) Mangle(sf reflect.StructField) ([]reflect.StructField,
 
 	switch k {
 	case reflect.Struct:
-		var err error
-		out, err = f.flattenStruct([]string{sf.Name}, prefixTag, sf)
-		if err != nil {
-			return out, err
+		// only flatten if it doesn't implement TextUnmarshaler
+		if !sf.Type.Implements(textMValue) {
```

Comment:
I think we also need to check `reflect.PtrTo(sf.Type).Implements(textMValue)`, since `Implements` doesn't check both the concrete type and the pointer-type. (in just about all cases `TextUnmarshal()` should take a pointer receiver)

Reply:
As discussed in slack, it's already a pointer when it gets there so checking to see if the concrete type implements it as well